llama-index-integrations/tools/llama-index-tools-agentql/llama_index/tools/agentql/agentql_browser_tool/base.py

```diff
@@ -0,0 +1,129 @@
+from typing import Optional
+
+from playwright.async_api import Browser as AsyncBrowser
+
+from llama_index.core.tools.tool_spec.base import BaseToolSpec
+
+from llama_index.tools.agentql.const import (
+    DEFAULT_EXTRACT_DATA_TIMEOUT_SECONDS,
+    DEFAULT_EXTRACT_ELEMENTS_TIMEOUT_SECONDS,
+    DEFAULT_WAIT_FOR_NETWORK_IDLE,
+    DEFAULT_INCLUDE_HIDDEN_DATA,
+    DEFAULT_INCLUDE_HIDDEN_ELEMENTS,
+    DEFAULT_RESPONSE_MODE,
+    REQUEST_ORIGIN,
+)
+from llama_index.tools.agentql.messages import (
+    QUERY_PROMPT_REQUIRED_ERROR_MESSAGE,
+    QUERY_PROMPT_EXCLUSIVE_ERROR_MESSAGE,
+)
+from llama_index.tools.agentql.utils import _aget_current_agentql_page
+
+
+class AgentQLBrowserToolSpec(BaseToolSpec):
+    """
+    AgentQL Browser Tool Spec.
+    """
+
+    spec_functions = [
+        "extract_web_data_from_browser",
+        "get_web_element_from_browser",
+    ]
+
+    def __init__(
+        self,
+        async_browser: AsyncBrowser,
+        timeout_for_data: Optional[int] = DEFAULT_EXTRACT_DATA_TIMEOUT_SECONDS,
```

Comment:
all these field don't have Optional type, they are optional by nature, because they have default values. But it's not allowed to pass None as their values